test/Microsoft.ML.TestFramework/BaseTestBaseline.cs

```diff
@@ -193,6 +193,9 @@ protected void Fail(bool relax, string fmt, params object[] args)
             {
                 _failures.Add(ex);
             }
+            catch when (relax)
```

Comment:
Why would we not want to report the callstack when relax is true? (More importantly what is the point of relax at all?)

Reply:
For these case, we only want to do deep compare of 2 score values and intend these 2 scores to be different.
when set this relax to true (from outside, it is allowMismatch), we want to compare values but not to fail the test (that is to say, after add Done() check).

---
In reply to: [375022748](https://github.com/dotnet/machinelearning/pull/4782#discussion_r375022748) [](ancestors = 375022748)

Reply:
Fail with a relax parameter seems really odd. This whole thing should be refactored differently.  (More notes below near the CompareNumbersWithTolerance function)

---
In reply to: [375064007](https://github.com/dotnet/machinelearning/pull/4782#discussion_r375064007) [](ancestors = 375064007,375022748)

Reply:
Yes, super strange